api.md

```diff
@@ -0,0 +1,230 @@
+# Table of contents
+- [API](#api)
+    - [Models](#models)
+        - [Project](#project)
+        - [Image](#image)
+        - [Developer](#developer)
+        - [Language](#language)
+    - [Endpoints](#endpoints)
+        - [/projects](#get-projects)
+        - [/project/id](#get-projectintid)
+        - [/developers](#get-developers)
+        - [/developer/id](#get-developerintid)
+
+# API
+[Back to Top](#table-of-contents)
+
+All endpoints will return a json response either acknowledging a POST request or with data relevant to a GET request. Errors should have a status in the 4XX or 5XX depending on the error and should include a 'message' json key with an adequate error message.
+
+## Models
+[Back to Top](#table-of-contents)
+
+All models should be described in the following sections. If implementation requires changes to these models please edit the corresponding information.
+
+### Project
+[Back to Top](#table-of-contents)
+
+Project is a model storing information about a project. It has a unique name, descriptions, icon, many languages, many contributors, a project status, a start year and an end year. This is the suggested django code which might need to be edited. (ID should not be included, it is auto generated)
+
```

Comment:
My idea for the project page was to have a video showcase using an iframe tag with a youtube embed link. That way the user wouldn't have to setup the repo for every project.  So it would need another field like ```youtube_link```.